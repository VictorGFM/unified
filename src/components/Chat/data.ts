import { Message } from '../Message/Message'

export const messages: Message[] = [
  { text: 'Some user message...', author: 'Victor Moraes' },
  { text: 'Some contact message...', author: 'Contact Name' },
]
