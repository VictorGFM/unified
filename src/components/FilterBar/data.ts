import { FiltersGroupsData } from './FilterBar'

export const Filters: FiltersGroupsData = {
  Posts: [
    {
      id: '1',
      label: 'Instagram Posts',
    },
    {
      id: '2',
      label: 'Twitter Posts',
    },
    {
      id: '3',
      label: 'Facebook Posts',
    },
  ],
  Groups: [
    {
      id: '4',
      label: 'Grupo dos Amigos',
    },
    {
      id: '5',
      label: 'Grupo do Trabalho',
    },
    {
      id: '6',
      label: 'Grupo da Família',
    },
    {
      id: '7',
      label: 'Grupo dos Canários',
    },
  ],
  Others: [
    {
      id: '8',
      label: 'Other Filter',
    },
    {
      id: '9',
      label: 'Other Filter',
    },
    {
      id: '10',
      label: 'Other Filter',
    },
  ],
}
