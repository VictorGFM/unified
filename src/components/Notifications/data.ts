import { AlfredoPhotoIcon, ContactPhotoIcon } from '../../assets'
import { Notification } from '../NotificationCard/NotificationCard'

export const notifications: Notification[] = [
  {
    id: '1',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    description: 'liked your tweet.',
    time: '1h',
    socialType: 'Twitter',
  },
  {
    id: '2',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '1h',
    socialType: 'Instagram',
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'shared your post.',
    time: '1h',
    socialType: 'Facebook',
  },
  {
    id: '4',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '1h',
    socialType: 'Instagram',
  },
  {
    id: '5',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '2h',
    socialType: 'Twitter',
  },
]
