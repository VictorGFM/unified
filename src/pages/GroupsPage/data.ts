import { GroupImage1, GroupImage2, GroupImage3, GroupImage4 } from '../../assets'
import { Group } from '../../components/GroupCard/GroupCard'

export const groups: Group[] = [
  { id: '1', title: 'Grupo dos Amigos', image: GroupImage1 },
  { id: '2', title: 'Grupo da Família', image: GroupImage2 },
  { id: '3', title: 'Grupo do Trabalho', image: GroupImage3 },
  { id: '4', title: 'Grupo dos Canários', image: GroupImage4 },
]
