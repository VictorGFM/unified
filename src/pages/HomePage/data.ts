import { NumberInteractions, PostInterface } from '../../components/Post/Post'
import { AlfredoPhotoIcon, AndromedaImage, ContactPhotoIcon, ProfilePhotoIcon } from '../../assets'

export const interactions: NumberInteractions[] = [
  {
    likes: 237,
    comments: 15,
    shares: 10,
  },
  {
    likes: 107,
    comments: 10,
    shares: 7,
  },
  {
    likes: 5,
    comments: 2,
    shares: 1,
  },
  {
    likes: 527,
    comments: 132,
    shares: 97,
  },
]

export const posts: PostInterface[] = [
  {
    id: '1',
    contactName: 'Victor Moraes',
    contactPhoto: ProfilePhotoIcon,
    contactGroup: 'Grupo da Família',
    postTime: '1h',
    socialType: 'Instagram',
    postDescription: 'Look at this galaxy! Its so beautiful...',
    postImage: AndromedaImage,
    numberInteractions: interactions[0],
  },
  {
    id: '2',
    contactName: 'Victor Moraes',
    contactPhoto: ProfilePhotoIcon,
    contactGroup: 'Grupo da Família',
    postTime: '3h',
    socialType: 'Twitter',
    postDescription:
      'So, this is a description of a Twitter post, but right after an Instagram post. All in the same place, awesome,' +
      ' isnt it?',
    numberInteractions: interactions[1],
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    contactGroup: 'Grupo dos Canários',
    postTime: '15h',
    socialType: 'Facebook',
    postDescription: 'Some post description...',
    numberInteractions: interactions[2],
  },
  {
    id: '4',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    contactGroup: 'Grupo dos Amigos',
    postTime: '21h',
    socialType: 'Twitter',
    postDescription: 'Au Au Au...',
    numberInteractions: interactions[3],
  },
]
