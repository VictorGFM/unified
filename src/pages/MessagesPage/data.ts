import { AlfredoPhotoIcon, ContactPhotoIcon } from '../../assets'
import { ContactInfo } from '../../components/ContactCard/ContactCard'

export const contacts: ContactInfo[] = [
  {
    id: '1',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    status: 'online',
    primaryInfo: 'Au Au Au...',
    secondaryInfo: 'Dec 5, 2021',
    contactNotifications: {
      total: 11,
      messages: {
        total: 11,
        instagram: 0,
        twitter: 10,
        facebook: 1,
      },
    },
  },
  {
    id: '2',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    status: 'online',
    primaryInfo: 'Some message...',
    secondaryInfo: 'Dec 15, 2021',
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    contactNotifications: {
      total: 41,
      messages: {
        total: 31,
        instagram: 17,
        twitter: 11,
        facebook: 3,
      },
      stories: {
        total: 5,
        instagram: 3,
        facebook: 2,
      },
    },
  },
  {
    id: '4',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '5',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '6',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '7',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '8',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '9',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '10',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '11',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '12',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '13',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '14',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '15',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '16',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '17',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '18',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '19',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '20',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '21',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '22',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '23',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
  {
    id: '24',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
  },
]
