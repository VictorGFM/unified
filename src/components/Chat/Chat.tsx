import { useState } from 'react'
import { FacebookIcon, InstagramIcon, TwitterIcon, InfoIcon, SendIcon } from '../../assets'
import { ContactInfo } from '../ContactCard/ContactCard'
import Message from '../Message'
import { SocialType } from '../SocialMediaBar/SocialMediaBar'
import SocialMediaButton from '../SocialMediaButton'
import { messages } from './data'
import {
  ChatContactCard,
  ChatDiv,
  ChatSocialMediaButton,
  InfoButton,
  InfoButtonImage,
  MessageDate,
  MessagesDiv,
  SendMessageBar,
  SocialBar,
  SocialMediaButtonsDiv,
} from './styles'

interface ChatProps {
  contact: ContactInfo
}

const Chat = ({ contact }: ChatProps) => {
  const [selectedSocial, setSelectedSocial] = useState('Instagram')

  const unreadMessages = contact.contactNotifications?.messages

  const handleClickButton = (socialType: SocialType) => {
    setSelectedSocial(socialType)
  }

  return (
    <ChatDiv>
      <SocialBar>
        <ChatContactCard contact={contact} />
        <SocialMediaButtonsDiv>
          <ChatSocialMediaButton
            icon={InstagramIcon}
            isSelected={selectedSocial === 'Instagram'}
            onClick={() => handleClickButton('Instagram')}
            unreadMessages={unreadMessages?.instagram}
          />
          <ChatSocialMediaButton
            icon={TwitterIcon}
            isSelected={selectedSocial === 'Twitter'}
            onClick={() => handleClickButton('Twitter')}
            unreadMessages={unreadMessages?.twitter}
          />
          <ChatSocialMediaButton
            icon={FacebookIcon}
            isSelected={selectedSocial === 'Facebook'}
            onClick={() => handleClickButton('Facebook')}
            unreadMessages={unreadMessages?.facebook}
          />
        </SocialMediaButtonsDiv>
        <InfoButton>
          <InfoButtonImage src={InfoIcon} />
        </InfoButton>
      </SocialBar>
      <MessagesDiv>
        <MessageDate>05/09/2021</MessageDate>
        <Message message={messages[0]} />
        <MessageDate>07/09/2021</MessageDate>
        <Message message={messages[1]} />
        <SendMessageBar icon={SendIcon} placeholder="Send a message..." />
      </MessagesDiv>
    </ChatDiv>
  )
}

export default Chat
