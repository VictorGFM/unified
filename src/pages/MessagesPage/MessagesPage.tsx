import Chat from '../../components/Chat'
import ContactMessages from '../../components/ContactMessages'
import { MessagesDiv, MessagesPageDiv } from './styles'
import { contacts } from './data'
import { useState } from 'react'
import { ContactInfo } from '../../components/ContactCard/ContactCard'

const MessagesPage = () => {
  const [selectedContact, setSelectedContact] = useState(contacts && contacts[0])

  const handleSelectContact = (contact: ContactInfo) => {
    setSelectedContact(contact)
  }

  return (
    <MessagesPageDiv>
      <MessagesDiv>
        <ContactMessages
          selectedContactID={selectedContact?.id}
          handleSelectContact={handleSelectContact}
        />
        {selectedContact && <Chat contact={selectedContact} />}
      </MessagesDiv>
    </MessagesPageDiv>
  )
}

export default MessagesPage
