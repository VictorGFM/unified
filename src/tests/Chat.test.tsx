import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'

import Chat from '../components/Chat'
import { contacts } from '../pages/MessagesPage/data'

describe('<Chat/>', () => {
  it('should show match snapshot', () => {
    const component = render(<Chat contact={contacts[0]} />)

    expect(component).toMatchSnapshot()
  })
})
