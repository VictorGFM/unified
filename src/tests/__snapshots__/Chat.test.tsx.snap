// Vitest Snapshot v1

exports[`<Chat/> > should show match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fEOsli bnYOuY"
      >
        <div
          class="sc-bjUoiL kTOCOT"
        >
          <div
            class="sc-dkzDqf iwBHmD sc-idiyUo hiHuRU"
          >
            <div
              class="sc-hKMtZM gUJqgj"
            >
              <img
                class="sc-eCYdqJ duHIXD"
                src="/src/assets/icons/AlfredoPhotoIcon.svg"
              />
              <div
                class="sc-jSMfEi Irfst"
              />
            </div>
            <div
              class="sc-gKXOVf fZYMRR"
            >
              <div
                class="sc-iBkjds fpwhgl"
              />
              <span
                class="sc-ftvSup keHLSh"
              >
                Au Au Au...
              </span>
              <span
                class="sc-papXJ fEZgDB"
              >
                Dec 5, 2021
              </span>
              <div
                class="sc-gsnTZi efXYKU sc-jqUVSM bWhrGW"
              >
                11
              </div>
            </div>
          </div>
          <div
            class="sc-dIouRR eOCFIi"
          >
            <div
              class="sc-breuTD ipeUcy sc-hHLeRK iWEOrF"
            >
              <img
                class="sc-ksZaOG cFmwJh"
                src="/src/assets/icons/InstagramIcon.svg"
              />
              <div
                class="sc-hAZoDl fiGYHm"
              />
            </div>
            <div
              class="sc-breuTD kqfbDV sc-hHLeRK iWEOrF"
            >
              <img
                class="sc-ksZaOG cFmwJh"
                src="/src/assets/icons/TwitterIcon.svg"
              />
              <div
                class="sc-gsnTZi efXYKU sc-fnykZs cIhCmT"
              >
                10
              </div>
            </div>
            <div
              class="sc-breuTD kqfbDV sc-hHLeRK iWEOrF"
            >
              <img
                class="sc-ksZaOG cFmwJh"
                src="/src/assets/icons/FacebookIcon.svg"
              />
              <div
                class="sc-gsnTZi efXYKU sc-fnykZs cIhCmT"
              >
                1
              </div>
            </div>
          </div>
          <div
            class="sc-dmRaPn bzpGvT"
          >
            <img
              class="sc-kgflAQ"
              src="/src/assets/icons/InfoIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-fLlhyt hvLRO"
        >
          <div
            class="sc-bBrHrO hsSQuB"
          >
            05/09/2021
          </div>
          <div
            class="sc-bczRLJ eDWSOr"
          >
            Some user message...
          </div>
          <div
            class="sc-bBrHrO hsSQuB"
          >
            07/09/2021
          </div>
          <div
            class="sc-bczRLJ cnuXqy"
          >
            Some contact message...
          </div>
          <div
            class="sc-kDDrLX kfOvYE sc-ivTmOn gdQqCi"
          >
            <input
              class="sc-iqcoie cnrCVV"
              placeholder="Send a message..."
            />
            <div
              class="sc-crXcEl hVwXjS"
            >
              <img
                class="sc-evZas ciNblN"
                src="/src/assets/icons/SendIcon.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fEOsli bnYOuY"
    >
      <div
        class="sc-bjUoiL kTOCOT"
      >
        <div
          class="sc-dkzDqf iwBHmD sc-idiyUo hiHuRU"
        >
          <div
            class="sc-hKMtZM gUJqgj"
          >
            <img
              class="sc-eCYdqJ duHIXD"
              src="/src/assets/icons/AlfredoPhotoIcon.svg"
            />
            <div
              class="sc-jSMfEi Irfst"
            />
          </div>
          <div
            class="sc-gKXOVf fZYMRR"
          >
            <div
              class="sc-iBkjds fpwhgl"
            />
            <span
              class="sc-ftvSup keHLSh"
            >
              Au Au Au...
            </span>
            <span
              class="sc-papXJ fEZgDB"
            >
              Dec 5, 2021
            </span>
            <div
              class="sc-gsnTZi efXYKU sc-jqUVSM bWhrGW"
            >
              11
            </div>
          </div>
        </div>
        <div
          class="sc-dIouRR eOCFIi"
        >
          <div
            class="sc-breuTD ipeUcy sc-hHLeRK iWEOrF"
          >
            <img
              class="sc-ksZaOG cFmwJh"
              src="/src/assets/icons/InstagramIcon.svg"
            />
            <div
              class="sc-hAZoDl fiGYHm"
            />
          </div>
          <div
            class="sc-breuTD kqfbDV sc-hHLeRK iWEOrF"
          >
            <img
              class="sc-ksZaOG cFmwJh"
              src="/src/assets/icons/TwitterIcon.svg"
            />
            <div
              class="sc-gsnTZi efXYKU sc-fnykZs cIhCmT"
            >
              10
            </div>
          </div>
          <div
            class="sc-breuTD kqfbDV sc-hHLeRK iWEOrF"
          >
            <img
              class="sc-ksZaOG cFmwJh"
              src="/src/assets/icons/FacebookIcon.svg"
            />
            <div
              class="sc-gsnTZi efXYKU sc-fnykZs cIhCmT"
            >
              1
            </div>
          </div>
        </div>
        <div
          class="sc-dmRaPn bzpGvT"
        >
          <img
            class="sc-kgflAQ"
            src="/src/assets/icons/InfoIcon.svg"
          />
        </div>
      </div>
      <div
        class="sc-fLlhyt hvLRO"
      >
        <div
          class="sc-bBrHrO hsSQuB"
        >
          05/09/2021
        </div>
        <div
          class="sc-bczRLJ eDWSOr"
        >
          Some user message...
        </div>
        <div
          class="sc-bBrHrO hsSQuB"
        >
          07/09/2021
        </div>
        <div
          class="sc-bczRLJ cnuXqy"
        >
          Some contact message...
        </div>
        <div
          class="sc-kDDrLX kfOvYE sc-ivTmOn gdQqCi"
        >
          <input
            class="sc-iqcoie cnrCVV"
            placeholder="Send a message..."
          />
          <div
            class="sc-crXcEl hVwXjS"
          >
            <img
              class="sc-evZas ciNblN"
              src="/src/assets/icons/SendIcon.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
